app/src/lib/menu-update.ts

```diff
@@ -145,8 +145,8 @@ function getMenuState(state: IAppState): Map<MenuIDs, IMenuItemState> {
     menuStateBuilder.setEnabled('compare-branch', isHostedOnGitHub && hasPublishedBranch)
 
     menuStateBuilder.setEnabled('view-repository-on-github', isHostedOnGitHub)
-    menuStateBuilder.setEnabled('push', !networkActionInProgress)
-    menuStateBuilder.setEnabled('pull', !networkActionInProgress)
+    menuStateBuilder.setEnabled('push', hasPublishedBranch && !networkActionInProgress)
```

Comment:
This is a good shout. This is how it is as of this PR:

 - for an unpublished repository, `Publish Repository` will also publish the `master` branch automagically
 - for a new branch after the repository has been published, you'll see this:

<img width="528" src="https://cloud.githubusercontent.com/assets/359239/26455634/1366ca12-416b-11e7-9b6f-a96c76402f13.png">

We don't look to see if any remotes have been defined for the repository - does that seem like a more relaxed approach here?